modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -134,9 +138,17 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
     }
 
   private val storeBlock: Pipe[F, FinalBlock, Unit] =
-    _.evalMap { fb =>
-      storages.block.store(fb.hashedBlock).handleErrorWith(e => logger.debug(e)("Block storing failed."))
-    }
+    _.evalTap(_ => semaphores.blockStoring.acquire).evalMap { fb =>
+      for {
+        lastSnapshotHeight <- storages.lastGlobalSnapshotStorage.getHeight.map(_.getOrElse(Height.MinValue))
+        _ <- if (lastSnapshotHeight.value < fb.hashedBlock.height.value)
+          storages.block.store(fb.hashedBlock).handleErrorWith(e => logger.debug(e)("Block storing failed."))
+        else
+          logger.debug(
+            s"Block can't be stored! Block height not above last snapshot height! block:${fb.hashedBlock.height} <= snapshot: $lastSnapshotHeight"
+          )
+      } yield ()
```

Comment:
what about a following syntax?
```scala
storages.lastGlobalSnapshotStorage.getHeight.map(_.getOrElse(Height.MinValue))
  .map(_.value < fb.hashedBlock.height.value)
  .ifM(
    storages.block.store(fb.hashedBlock).handleErrorWith(...),
    logger.debug(s"Block can't be stored - height above last snapshot height.")
  )
```

Reply:
Initially that's how I implemented it, but then when writing a debug message I wanted to supply the context of snapshot height and block height, so I switched to for comprehension. If we are ok with more general debug message I can go with this notation.

Reply:
Hmm, for me, using a flatMap here is still more clear, but up to you. It's a detail